corehq/apps/app_manager/suite_xml/post_process/workflow.py

```diff
@@ -306,6 +306,29 @@ def frame_children_for_module(module_, include_user_selections=True):
 
                 stack_frames.append(StackFrameMeta(link.xpath, frame_children, current_session=source_form_datums))
 
+            if form.post_form_workflow_backup:
+                if form.post_form_workflow_backup == WORKFLOW_ROOT:
+                    stack_frames.append(StackFrameMeta(None, [], allow_empty_frame=True))
+                elif form.post_form_workflow_backup == WORKFLOW_MODULE:
+                    frame_children = frame_children_for_module(module, include_user_selections=False)
+                    stack_frames.append(StackFrameMeta(None, frame_children))
+                elif form.post_form_workflow_backup == WORKFLOW_PARENT_MODULE:
+                    root_module = module.root_module
+                    frame_children = frame_children_for_module(root_module)
+                    stack_frames.append(StackFrameMeta(None, frame_children))
+                elif form.post_form_workflow_backup == WORKFLOW_PREVIOUS:
+                    frame_children = self.helper.get_frame_children(form)
+
+                    # since we want to go the 'previous' screen we need to drop the last
+                    # datum
+                    last = frame_children.pop()
+                    while isinstance(last, WorkflowDatumMeta) and not last.requires_selection:
+                        # keep removing last element until we hit a command
+                        # or a non-autoselect datum
+                        last = frame_children.pop()
+
+                    stack_frames.append(StackFrameMeta(None, frame_children))
```

Comment:
Instead of just copying this from above can you not extract it to a function?

For this to work correctly as a fall-back option you'll need to set the `if_clause` in the `StackFrameMeta` to be the inverse of all the `link.xpath` expressions.